src/_P092_DLbus.ino

```diff
@@ -809,10 +818,23 @@ void P092_Pin_changed(void) {
     else if (DLbus_time_diff > DLbus_MaxPulseWidth)         val |= DLbus_FlagBetweenDoubleSingleWidth;    // between double and single pulse width
     else if (DLbus_time_diff < DLbus_MinPulseWidth)         val |= DLbus_FlagShorterThanSingleWidth;      // shorter then single pulse width
     else val |= DLbus_FlagSingleWidth;                                   // single pulse width
-    DLbus_BitStream[DLbus_pulse_count] = val;                            // set bit
-    DLbus_pulse_count++;
-    P092_receiving = (DLbus_pulse_count < DLbus_pulse_number);     // stop P092_receiving when data frame is complete
-    P092_AllBitsReceived = !P092_receiving;
+    if (DLbus_pulse_count < 2) {
+      // check if sync is received
+      if (val & DLbus_FlagLongerThanTwiceDoubleWidth) {
+        // sync received
+        *DLbus_PtrChangeBitStream = !(val & 0x01);
```

Comment:
There is no guarantee this pointer is valid when this function is called as the pointer is not guaranteed to be set when the interrupt is attached.
Also, why do you use a pointer to an object which is always present and never increase the pointer?
It would make sense to use a pointer when you use it as an iterator through the array, or when the array has to be created at init for example.

Reply:
I use this pointer as volatile because it is called in the ISR call.
The array the pointer points to is not longer volatile, that means not stored in th IRAM section.
This pointer is valid because it is set before the `P092_receiving` is set tu `true` (line 700)!

Reply:
Not 100% sure how the compiler will use the array here.
But I understand now why you did it.

I missed the `P092_receiving` in line 700, but you are right. This then acts as a guard for the initialized pointer.

Reply:
It's line 701 in the changed file.

Reply:
Is there a chance to see the assembler code created by the compiler?

Reply:
I'm sure there is, but even then I don't know how to interpret the allocated memory ranges to see if it will cause an issue or not.
The core libs have some checks for missing attributes for ISR related code, but I am not sure if it also checks (or need to check) for memory access address ranges.
I guess you need to check stability over time.

Reply:
So far the changed code runs stable for about 90 minutes.
There are no sync information of the DLbus missed.
The received bit changes can be decoded (manchester code) and the CRC check is always passed.
```
4507343: PLUGIN_092_READ
4507344: Get Sensor1: 21.7
4507439: PLUGIN_092_READ
4507439: Get Sensor2: 29.2
4507480: PLUGIN_092_READ
4507480: Get DigitalOutput1: 0.0
4507521: PLUGIN_092_READ
4507522: Get AnalogOutput1: 0.0
4508736: WD : Uptime 75 ConnectFailures 0 FreeMem 18192 WiFiStatus 3
4512499: SYS : 75.00,15.30,18296.00,3376.00
4519265: P092_receiving ...
4521266: Receive stopped.
4521267: DLbus_ChangeBitStream copied.
4521267: Processing...
4521268: Invert bit stream...
4521269: StartBit: 36 / EndBit: 413
4521271: Check CRC...
4521271: Received data OK
4538736: WD : Uptime 76 ConnectFailures 0 FreeMem 18192 WiFiStatus 3
4552266: P092_receiving ...
4554266: Receive stopped.
4554267: DLbus_ChangeBitStream copied.
4554267: Processing...
4554268: Invert bit stream...
4554269: StartBit: 36 / EndBit: 414
4554271: Check CRC...
4554271: Received data OK
4567366: PLUGIN_092_READ
4567367: Get Sensor1: 21.7
4567488: PLUGIN_092_READ
4567489: Get Sensor2: 29.0
4567538: PLUGIN_092_READ
4567538: Get DigitalOutput1: 0.0
4567578: PLUGIN_092_READ
4567578: Get AnalogOutput1: 0.0
4568743: WD : Uptime 76 ConnectFailures 0 FreeMem 16568 WiFiStatus 3
```

Reply:
Well if there is an issue with ISR code, caused by variables not defined like they should, then it will simply crash with some kind of exception.